site/content/docs/03-run-time.md

```diff
@@ -50,6 +50,8 @@ If a function is returned from `onMount`, it will be called when the component i
 </script>
 ```
 
+> This behaviour will only work when the function passed to `onMount` *synchronously* returns a value. `async` functions always return a `Promise`, and as such cannot *synchronously* return a function.
```

Comment:
Do we want to maybe say "returns a function" rather than "returns a value"? The only thing I'd worry with about that is the ends of the two sentences sounding a bit repetitive.

Reply:
The reason I've said `value` in the first instance, is that I wouldn't consider this:

```
const myOnDestroy = function () { disposeStuff() } 
onMount(async () => new Promise(r => r(myOnDestroy))
```

as returning a function, more returning a value. I'd want to cover people literally using promises as return values within async functions.

Reply:
That's fair. This seems like a good explanation then, and there'll at least be some place in the docs to point people to about this now.